src/hooks/useMyAxios.ts

```diff
@@ -1,17 +1,13 @@
-import Axios, { AxiosInstance, AxiosRequestConfig } from 'axios'
+import Axios, { AxiosInstance } from 'axios'
 import { makeUseAxios } from 'axios-hooks'
-import { useCallback } from 'react'
-
-const BASE_URL = 'http://localhost:2337/' // TODO
-const BASIC_REQUEST_CONFIG: AxiosRequestConfig = {
-  baseURL: BASE_URL,
-}
+import { useCallback, useMemo } from 'react'
 
 const useMyAxios = () => {
   // const axiosInstance: AxiosInstance = Axios.create(BASIC_REQUEST_CONFIG)
-  const axiosInstance: AxiosInstance = Axios.create()
+  const axiosInstance: AxiosInstance = useMemo(() => { return Axios.create() }, [])
```

Comment:
Does `useMemo` with an empty array do anything that its omission would not?

Reply:
Omitting the [] would cause the `Axios.create()` call to execute on every render, which then causes `useEffect`s to re-fire. 

Each use of the `axiosInstance` from this creates a new usable object with the correct URL to make the fetch call. I did verify that the token gets updated on the request when the timeout occurs after 15 minutes and doesn't force any unnecessary refreshes.

Reply:
Sorry, I meant omitting `useMemo`, not omitting the array.

Reply:
Interesting. I think I need to work on my mental model of how exactly `useMemo` works.